provider/resource.go

```diff
@@ -252,6 +252,42 @@ func stringValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []st
 	return ""
 }
 
+// boolValue takes a bool value from a property map, then from environment vars; defaults to false
+func boolValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []string) bool {
+	val, ok := vars[prop]
+	if ok && val.IsBool() {
+		return val.BoolValue()
+	}
+	for _, env := range envs {
+		val, ok := os.LookupEnv(env)
+		if ok && val == "true" {
+			return true
+		}
+	}
+	return false
+}
+
+// arrayValue takes an array value from a property map, then from environment vars; defaults to an empty array
+func arrayValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []string) []string {
+	val, ok := vars[prop]
+	var vals []string
+	fmt.Println(vals)
+	if ok && val.IsArray() {
+		for _, v := range val.ArrayValue() {
+			vals = append(vals, v.StringValue())
+		}
+		return vals
+	}
+
+	for _, env := range envs {
+		val, ok := os.LookupEnv(env)
+		if ok {
+			return strings.Split(val, ";")
```

Comment:
do we need to handle "; " with a space?

Reply:
I really don't think so: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/provider/provider.go#L328